File: TUnit.Core.SourceGenerator.Tests/ClassConstructorTest.cs
using TUnit.Core.SourceGenerator.Tests.Options;

namespace TUnit.Core.SourceGenerator.Tests;

internal class ClassConstructorTest : TestsBase
{
    [Test]
    public Task Test() => RunTest(Path.Combine(Git.RootDirectory.FullName,
            "TUnit.TestProject",
            "ClassConstructorTest.cs"),
        new RunTestOptions
        {
            AdditionalFiles =
            [
                Path.Combine(Git.RootDirectory.FullName,
                    "TUnit.TestProject",
                    "DummyReferenceTypeClass.cs"),
                Path.Combine(Git.RootDirectory.FullName,
                    "TUnit.TestProject",
                    "DependencyInjectionClassConstructor.cs")
            ]
        },
        async generatedFiles =>
        {
            });
}

File: TUnit.TestProject.VB.NET/ClassConstructorTest.vb
Imports TUnit.Core

<ClassConstructor(GetType(DependencyInjectionClassConstructor))>
Public Class ClassConstructorTest

    Public Sub New(dummyReferenceTypeClass As DummyReferenceTypeClass)
        Me.DummyReferenceTypeClass = dummyReferenceTypeClass
    End Sub

    Public ReadOnly Property DummyReferenceTypeClass As DummyReferenceTypeClass

    <Test>
    Public Sub Test()
        ' Test logic here
    End Sub

End Class